cmd/pomerium-cli/cache_test.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// creating Mock Cache operations
+type MockCache struct {
+	mock.Mock
+}
+
+// mock func for cache.ExecCredentialsPath()
+func (m *MockCache) ExecCredentialsPath() (string, error) {
+	args := m.Called()
+	return args.String(0), args.Error(1)
+}
+
+func TestClearAllCachedCredentials(t *testing.T) {
+	filePath := t.TempDir()
+	//mock object
+	mockCache := new(MockCache)
+
+	//set up dir expectation for ExecCredentialsPath
+	mockCache.On(" ExecCredentialsPath").Return(filePath+"/tmp/cache", nil)
```

Comment:
You've correctly identified that ExecCredentialsPath() makes this behavior difficult to test.

However I don't think this mock object is effective at swapping out a different path. There are some early returns below, but if I change those, I see that `clearAllCachedCredentials()` uses the real ExecCredentialsPath() rather than this mock path.